### PR TITLE
Update JobContext

### PR DESCRIPTION
**Pull request by riversand963**
In the past, we assume that a job modifies a single column family. Therefore, a job can create at most one superversion since each superversion corresponds to one column family. This assumption leads to the fact that a `JobContext` has only one member variable called `superversion_context`.
Now we want to support group flush of column families, indicating that each job can create multiple superversions. Therefore, we need to make the following change to accommodate this new feature.

Add a vector of `SuperVersionContext` to `JobContext` to support installing
superversions for multiple column families in one job context.

This PR is a subset of [PR 3752](https://github.com/facebook/rocksdb/pull/3752).

Test plan:
```
$ make clean && make -j16 all check
```

All tests should pass.

### CODE REVIEW

File: db/job_context.h [hunk 1]
```diff
@@ -142,8 +149,12 @@ struct JobContext {
   // a list of memtables to be free
   autovector<MemTable*> memtables_to_free;
 
+  // context for installing superversion for one column family
   SuperVersionContext superversion_context;
 
+  // contexts for installing superversions for multiple column families
+  std::vector<SuperVersionContext> superversion_contexts;
```
Comment (ajkr) [inline db/job_context.h]: who sets superversion_contexts?

Comment (riversand963) [reply db/job_context.h]: In [PR 3752](https://github.com/facebook/rocksdb/pull/3752), we have an overloaded `InstallSuperVersionAndScheduleWork` which calls `NewSuperVersion()` for each element of `superversion_contexts`.

Comment (ajkr) [reply db/job_context.h]: In the `atomic_flush_=false` case, can we have `superversion_contexts` be a vector of length one to eliminate the redundant variable?

Comment (riversand963) [reply db/job_context.h]: Should be doable. Will update code.

Comment (riversand963) [reply db/job_context.h]: Done.

**Pull request closed without merging**